File: ios/Classes/entity/V2TimPermissionGroupOperationResult.swift
import Foundation
import ImSDK_Plus

/// 自定义消息响应实体
class V2TimPermissionGroupOperationResult: V2TIMPermissionGroupOperationResult {

    /// 根据对象获得字典对象
    public static func getDict(info: V2TIMPermissionGroupOperationResult) -> [String: Any] {
        var result: [String: Any] = [:];
        result["permissionGroupID"] = info.permissionGroupID;
        result["resultCode"] = info.resultCode;
        result["resultMessage"] = info.resultMsg;
       
         return result;
    }
    init(dict: [String: Any]) {
        super.init();
            
        //  self.groupID = (dict["groupID"] as? String);
		
		//  if let customData = dict["customData"] as? String {
		//  	self.customData = customData;
		//  }
		//  if let groupPermission = dict["groupPermission"] as? UInt64 {
		//  	self.groupPermission = groupPermission;
		//  }
		//  if let permissionGroupID = dict["permissionGroupID"] as? String {
		//  	self.permissionGroupID = permissionGroupID;
		//  }
		//  if let permissionGroupName = dict["permissionGroupName"] as? String {
		//  	self.permissionGroupName = permissionGroupName;
		//  }
    }
    public static func getListDict(infos: [V2TIMPermissionGroupOperationResult]) -> [[String: Any]] {
       
        var res: [[String: Any]] = []
        for info in infos {
            res.append(V2TimPermissionGroupOperationResult.getDict(info: info))
        }
        return res;
    }
}

File: lib/models/v2_tim_permission_group_operation_result.dart
class V2TimPermissionGroupOperationResult {
  int resultCode;
  String resultMessage;
  String permissionGroupID;
  V2TimPermissionGroupOperationResult({
    required this.resultCode,
    required this.resultMessage,
    required this.permissionGroupID,
  });
  Map<String, dynamic> toJson() {
    return Map<String, dynamic>.from({
      'resultCode': resultCode,
      'resultMessage': resultMessage,
      'permissionGroupID': permissionGroupID,
    });
  }

  static V2TimPermissionGroupOperationResult fromJson(
      Map<String, dynamic> json) {
    return V2TimPermissionGroupOperationResult(
      resultCode: json['resultCode'] ?? 0,
      resultMessage: json['resultMessage'] ?? "",
      permissionGroupID: json['permissionGroupID'] ?? "",
    );
  }
  String toLogString() {
    String res = "resultCode:$resultCode|resultMessage:$resultMessage|permissionGroupID:$permissionGroupID";
    return res;
  }
}
